src/tool_parsecfg.c

```diff
@@ -335,39 +343,21 @@ static const char *unslashquote(const char *line, char *param)
 
 /*
  * Reads a line from the given file, ensuring is NUL terminated.
- * The pointer must be freed by the caller.
- * NULL is returned on an out of memory condition.
+ * FALSE is returned on an out of memory condition.
  */
-static char *my_get_line(FILE *fp)
+static bool my_get_line(FILE *fp, struct curlx_dynbuf *db)
 {
   char buf[4096];
   char *nl = NULL;
-  char *line = NULL;
 
   do {
     if(NULL == fgets(buf, sizeof(buf), fp))
-      break;
-    if(!line) {
-      line = strdup(buf);
-      if(!line)
-        return NULL;
-    }
-    else {
-      char *ptr;
-      size_t linelen = strlen(line);
-      ptr = realloc(line, linelen + strlen(buf) + 1);
-      if(!ptr) {
-        Curl_safefree(line);
-        return NULL;
-      }
-      line = ptr;
-      strcpy(&line[linelen], buf);
-    }
-    nl = strchr(line, '\n');
+      return FALSE;
+    nl = strchr(buf, '\n');
+    if(curlx_dyn_add(db, buf))
+      /* failure */
+      return FALSE;
   } while(!nl);
```

Comment:
```suggestion
  } while(!feof(fp) && !strchr(buf, '\n'));
```
Otherwise the last line is ignored if it doesn't end in newline (fgets fails and so getline returns false). And should dyn buf be reset on FALSE condition?

Reply:
Oh very good point!